### ISSUE TITLE
nil Item ID

### ISSUE THREAD

**Issue by descention** [descention/Compat#1]
Having some issues with a nil item id. TSM Version 3.6.43

```
20x TradeSkillMaster\CoreAPI\Items-Items.lua:625: attempt to concatenate local 'itemId' (a nil value)
TradeSkillMaster\CoreAPI\Items-Items.lua:625: in function '?'
TradeSkillMaster\CoreAPI\Threading.lua:489: in function <TradeSkillMaster\CoreAPI\Threading.lua:477>

Locals:
self = <unnamed> {
 0 = <userdata>
}
event = "GET_ITEM_INFO_RECEIVED"
shouldUnregister = true
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 (null) = <table> {
 }
 (null) = <table> {
 }
 (null) = <table> {
 }
 (null) = <table> {
 }
 (null) = <table> {
 }
}
(for control) = <table> {
}
_ = <table> {
}
thread = <table> {
 sleepTime = 0.1
 stats = <table> {
 }
 messages = <table> {
 }
 co = <no value>
 endTime = 87878139.869631
 _name = "ITEM_INFO"
 state = "SLEEPING"
 id = <table> {
 }
 events = <table> {
 }
 caller = "TradeSkillMaster/CoreAPI/Items.lua:361"
 priority = 0.1
 obj = <table> {
 }
}
private = <table> {
 RunScheduler = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:363
 ProcessEvent = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:477
 threadSort = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:359
 CheckWaitFunctionResult = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:352
 RunThread = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:319
 ThreadDefaults = <table> {
 }
 GetThreadFunctionWrapper = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:498
 ThreadPrototype = <table> {
 }
 frame = <unnamed> {
 }
 threads = <table> {
 }
 GetCurrentThreadPosition = <function> defined @TradeSkillMaster\CoreAPI\Threading.lua:525
}```

**Comment by descention** [comment]
```90x TradeSkillMaster\CoreAPI\Items-Items.lua:301: attempt to compare nil with number
TradeSkillMaster\CoreAPI\Items-Items.lua:301: in function `IsDisenchantable'
...faceTradeSkillMaster\CoreAPI\Conversions.lua:143: in function `GetValue'
TradeSkillMaster\TradeSkillMaster-v3.6.43.lua:521: in function `LoadTooltip'
TradeSkillMaster\Private\Tooltips.lua:91: in function `callback'
...rfaceTradeSkillMaster\Private\TooltipLib.lua:143: in function <...rfaceTradeSkillMaster\Private\TooltipLib.lua:122>
...rfaceTradeSkillMaster\Private\TooltipLib.lua:83: in function <...rfaceTradeSkillMaster\Private\TooltipLib.lua:82>
[C]: ?
...rfaceTradeSkillMaster\Private\TooltipLib.lua:95: in function `SetLootItem'
FrameXML\LootFrame.lua:270: in function `LootItem_OnEnter'
[string "*:OnUpdate"]:2: in function <[string "*:OnUpdate"]:1>

Locals:
self = <table> {
 GetSourceItems = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:101
 GetData = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:40
 GetTargetItemsByMethod = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:126
 GetValue = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:139
 GetTargetItemNames = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:64
 GetTargetItemByName = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:44
 Add = <function> defined @TradeSkillMaster\CoreAPI\Conversions.lua:20
}
sourceItem = "i:55973:0"
customPrice = "dbmarket"
method = "disenchant"
TSM = <table> {
 Operations = <table> {
 }
 SetDefaultModuleLibraries = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:398
 _author = "Sapu94"
 DeleteCustomPriceSource = <function> defined @TradeSkillMaster\CoreAPI\Prices.lua:56
 EnableModule = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:363
 modules = <table> {
 }
 baseName = "TradeSkillMaster"
 IterateEmbeds = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:473
 CreateGroupTree = <function> defined @TradeSkillMaster\GUI\TSMFrames\GroupTree.lua:291
 SecureHookScript = <function> defined @TradeSkillMaster\Libs\AceHook-3.0\AceHook-3.0.lua:410
 GetAuctionPlayer = <function> defined @TradeSkillMaster\GUI\AuctionFrame.lua:45
 RawHookScript = <function> defined @TradeSkillMaster\Libs\AceHook-3.0\AceHook-3.0.lua:395
 _desc = "Core addon for the TradeSkillMaster suite. Does nothing without modules installed."
 Printf = <function> defined @TradeSkillMaster\CoreAPI\Modules.lua:147
 SetDefaultModulePrototype = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:440
 NO_SOUND_KEY = "TSM_NO_SOUND"
 LDBIcon = <table> {
 }
 defaultModuleState = true
 IsEnabled = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:482
 LOG_INFO = <function> defined @TradeSkillMaster\Debug\Logging.lua:376
 Delay = <table> {
 }
 OnTSMDBShutdown = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:418
 operations = <table> {
 }
 RegisterModule = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:346
 toggleBankUI = <function> defined @TradeSkillMaster\GUI\BankUI.lua:307
 ShowConfigError = <function> defined @TradeSkillMaster\Debug\ErrorHandler.lua:79
 RegisterMessage = <function> defined @BugSack\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:118
 UnregisterMessage = <function> defined @BugSack\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:181
 LOG_ERR = <function> defined @TradeSkillMaster\Debug\Logging.lua:384
 PrintVersion = <function> defined @TradeSkillMaster\TradeSkillMaster.lua:757
 hooks = <table> {
 }
 getBankTabs = <function> defined @TradeSkillMaster\GUI\BankUI.lua:325
 Groups = <table> {
 }
 UnregisterAllEvents = <function> defined @BugSack\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:202
 db = <table> {
 }
 LOG_RAISE_STACK = <function> defined @TradeSkillMaster\Debug\Logging.lua:371
 AnalyticsEvent = <function> defined @TradeSkillMaster\CoreAPI\Analytics.lua:52
 GetArgs = <function> defined @TradeSkillMaster\Libs\AceConsole-3.0\AceConsole-3.0.lua:144
 SetEnabledState = <function> defined @TradeSkillMaster\Libs\AceAddon-3.0\AceAddon-3.0.lua:455
 Hook = <function> defined @TradeSkillMaster\Libs\AceHook-3.0\AceHook-3.0.lua:274
 enabledState = true
 UpdateDesign = <function> defined @TradeSkillMaster\GUI\Design.lua:75
 Registe```

**Comment by descention** [comment]
More stacktrace
```
Addon: TradeSkillMaster
Message: Interface\AddOns\TradeSkillMaster\CoreAPI\Items.lua:624: attempt to concatenate global 'itemId' (a nil value)
Date: 01/15/23 16:24:41
Client: 5.4.8
Locale: enUS
Combat: nil
Stack:
    TSM\CoreAPI\Threading.lua:502: <Threading.lua:499>
TSM Thread Info:
    {
      ["TradeSkillMaster_Accounting/Modules/GoldTracker.lua:24"]={
        state="READY",
        isImmortal=true,
        priority=0.3,
        caller="TradeSkillMaster_Accounting/Modules/GoldTracker.lua:24",
        threadId="table: 0000000023FDA610"
      },
      ["TradeSkillMaster/CoreAPI/Delay.lua:82"]={
        state="READY",
        isImmortal=true,
        priority=0.4,
        caller="TradeSkillMaster/CoreAPI/Delay.lua:82",
        threadId="table: 000000001F291A50"
      },
      ITEM_INFO={
        state="RUNNING",
        stats={
          realTime=0.20329999970272,
          cpuTime=0,
          cpuPct="0.0%",
          overTimeCount=0,
          numYields=0
        },
        caller="TradeSkillMaster/CoreAPI/Items.lua:361",
        priority=0.1,
        threadId="table: 000000001F034280"
      },
      ["TradeSkillMaster/CoreAPI/Inventory.lua:348"]={
        state="READY",
        isImmortal=true,
        priority=0.3,
        caller="TradeSkillMaster/CoreAPI/Inventory.lua:348",
        threadId="table: 000000002180D100"
      },
      ["TradeSkillMaster/CoreAPI/Sync.lua:179"]={
        state="READY",
        caller="TradeSkillMaster/CoreAPI/Sync.lua:179",
        priority=0.5,
        threadId="table: 0000000023FDA480"
      }
    }
TSM Debug Log:
    2023/01/15 16:24:40.660 [Vendoring:INFO:69] Registered with TSM!
    2023/01/15 16:24:40.660 [Mailing:INFO:95] Registered with TSM!
    2023/01/15 16:24:40.658 [TSM (Core):ERR:499] Invalid names (1, 0)
    2023/01/15 16:24:40.201 [Shopping:INFO:84] Registered with TSM!
    2023/01/15 16:24:40.168 [Accounting:INFO:153] Registered with TSM!
    2023/01/15 16:24:40.099 [TSM (Core):INFO:415] Registered with TSM!
Addons:
    ACP (r192)
    !BugGrabber (r203-release)
    BugSack (r278-release)
    Compat (v1)
    TSM (v3.6.43)
    TSM_Accounting (v3.0.20)
    TSM_Mailing (v3.0.18)
    TSM_Shopping (v3.4.7)
    TSM_Vendoring (v3.0.7)
```

**Comment by descention** [comment]
This is partially addressed with fixes to GetItemInfoInstant 7c07b6ce5269072df3475c6bd3bac2d59436523b. It doesn't fix it completely though.

**Comment by descention** [comment]
This is mitigated by 09887401e95260ecb48b48d1c0af8acd41f69553. I'm planning to address GetItemInfoInstant at a later date and is tracked by #5 .
